src/WebApi.OutputCache.V2/CacheOutputAttribute.cs

```diff
@@ -242,12 +255,12 @@ public override void OnActionExecuting(HttpActionContext actionContext)
                         _webApiCache.Add(baseKey, string.Empty, cacheTime.AbsoluteExpiration);
                         _webApiCache.Add(cachekey, content, cacheTime.AbsoluteExpiration, baseKey);
```

Comment:
Sorry to jump into this discussion without invitation, but I thought on something similar: when a client issues a `no-cache` is actually saying "bypass whatever you have cached", but not saying that "you should remove your cached data".
I implemented something similar in a project and our approach was much simpler: if the client sent a `no-cache` it was exactly the same as if the controller said the results shouldn't be cached. And to implement so we overloaded the  `IsCachingAllowed`, adding a condition similar to what's evaluated in @abpatel `IsNoCacheHeaderInRequest`.
If you like, I can create a PR with that.